docs/book/src/reference/metrics.md

```diff
@@ -59,6 +60,16 @@ Follow the steps below to export the metrics using the Prometheus Operator:
 - ../prometheus
 ```
 
+3. To apply the necessary RBAC resource for Prometheus operator, uncomment the
+line `- ../prometheus_rbac` in the `config/default/kustomization.yaml`. This
+will create a `Role` and `RoleBinding` resources which enables access to the
+metrics.
+
+```yaml
+# [PROMETHEUS RBAC] To give Prometheus service account necessary permissions, uncomment all sections with 'PROMETHEUS RBAC'.
+- ../prometheus_rbac
+```
```

Comment:
Cluster admins have many ways to configure these permissions and IMO configuring the Prometheus permissions would not be part of the responsibility of an Operator and it would probably hurt concepts like single responsibility.  

We also, address the scenario where someone is trying to check it locally and is unable to because of the lack the permissions via the docs: 

(https://master.book.kubebuilder.io/reference/metrics.html)

<img width="852" alt="Screenshot 2022-09-02 at 07 18 12" src="https://user-images.githubusercontent.com/7708031/188071616-0a88b43a-1b9d-453f-ac3e-65f6a5f0f1c6.png">

And as described cluster admins have many ways to configure it. 

My vote here is we close this one as defered.

Reply:
I agree. Exposing metrics and providing `servicemonitor` tmpl are fairly enough for operator developing.
Another concern is: introducing additional RBAC here may potentially bind kubebuilder with prometheus-opeartor, which seems not necessary.
How users running their prometheus instances may very, `prometheus-operator`/`kube-prometheus` is an awesome solution but not the only one.